File: java/swea8931.java
package p0127;

import java.util.Scanner;
import java.util.Stack;

public class swea8931 {
	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		int tc = sc.nextInt();
		for (int t = 1; t <= tc; t++)
		{
			int k =sc.nextInt(),sum = 0;
			Stack<Integer> s = new Stack<>();
			for(int i = 0 ; i < k ; i++)
			{
				int x = sc.nextInt();
				if(x != 0)
				{
					s.push(x);
					sum +=x;
				}
				else
					sum -= s.pop();
			}
			System.out.println("#"+t+" "+sum);
		}
		
	}
}

File: c++/swea8931.cpp
#include<cstdio>
#include<stack>
using namespace std;

int main()
{
	int tc;
	scanf("%d", &tc);
	for (int t = 1; t <= tc; t++)
	{
		int k, sum = 0;
		scanf("%d", &k);
		stack<int> s;
		while (k--)
		{
			int x;
			scanf("%d", &x);
			if (x != 0)
			{
				s.push(x);
				sum += x;
			}
			else
			{
				sum -= s.top();
				s.pop();

			}
		}
	}
}